src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTag.cs

```diff
@@ -18,18 +32,87 @@ namespace Microsoft.CodeAnalysis.Editor.InlineParameterNameHints
+        private readonly ITextView _textView;
+        private SnapshotSpan _span;
+        private readonly SymbolKey _key;
+        private readonly IThreadingContext _threadingContext;
+        public Lazy<IStreamingFindUsagesPresenter> _streamingPresenter;
 
         /// <summary>
         /// Creates the UIElement on call
         /// Uses PositionAffinity.Successor because we want the tag to be associated with the following character
         /// </summary>
-        /// <param name="text">The name of the parameter associated with the argument</param>
-        public InlineParameterNameHintsTag(string text, double lineHeight, TextFormattingRunProperties format)
-            : base(CreateElement(text, lineHeight, format), removalCallback: null, PositionAffinity.Successor)
+        /// <param name="adornment">The adornment that we are creating</param>
+        /// <param name="toolTipService">The service being used to create the tooltip and match the UI</param>
+        /// <param name="textView">The view of the editor</param>
+        /// <param name="span">The span that has the location of the hint</param>
+        /// <param name="key">The symbolkey associated with each parameter</param>
+        private InlineParameterNameHintsTag(FrameworkElement adornment,
+                                           IToolTipService toolTipService, ITextView textView,
+                                           SnapshotSpan span, SymbolKey key, IThreadingContext threadingContext,
+                                           Lazy<IStreamingFindUsagesPresenter> streamingPresenter)
+            : base(adornment, removalCallback: null, PositionAffinity.Successor)
         {
+            _streamingPresenter = streamingPresenter;
+            _textView = textView;
+            _key = key;
+            _toolTipService = toolTipService;
+            _span = span;
+            _threadingContext = threadingContext;
+            adornment.ToolTip = "Test";
+            adornment.ToolTipOpening += Border_ToolTipOpening;
         }
 
-        private static UIElement CreateElement(string text, double lineHeight, TextFormattingRunProperties format)
+        public static InlineParameterNameHintsTag Create(string text, double lineHeight, TextFormattingRunProperties format, IToolTipService toolTipService, ITextView textView,
+                                          SnapshotSpan span, SymbolKey key, IThreadingContext threadingContext, Lazy<IStreamingFindUsagesPresenter> streamingPresenter)
+        {
+            return new InlineParameterNameHintsTag(CreateElement(text, lineHeight, format), toolTipService, textView, span, key, threadingContext, streamingPresenter);
+        }
+
+        public async Task<IReadOnlyCollection<object>> CreateDescriptionAsync()
+        {
+            var cancellationToken = new CancellationToken();
+            var document = _textView.TextBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
```

Comment:
This could come back null. Let's `#nullable enable` the file and then you'll see all the places you need to add null checks for.